### PR TITLE
#100 issue fix

### LINKED ISSUE #100: Not possible to update an existing issue?
I've spent about the last six hours trying to figure out how to use the JIRA API from Ruby, using this gem, to create, update, and comment on an issue in JIRA.  Simple, right?  NO.  This gem is so poorly documented that it took hours just to log into our JIRA account with it because the secret of using :context => '' is not mentioned anywhere and I had to figure it out myself.  Then I had to figure out how to find the IDs for projects and issue types by inspecting the HTML from atlassian.net and I was finally able to post an issue.

Updating an existing issue, however, is a complete mystery.  I'm guessing that it will be like the other issues: that it will work if I can find some sample code somewhere.  The sample code in the documentation for updating records absolutely does not work.  Now I'm just throwing things against the wall to see if anything sticks.  I seriously think that I would have been better off pulling out Capybara and writing a web bot to do this instead of trying to use this terrible API gem.

Here are a few examples of failed attempts to post an update to an existing issue.  I get a 400 response and no error messages in the issue's data structure.

```
client = JIRA::Client.new(options)
issue = client.Issue.find('ENG-23')
begin
  issue.save!(
      {
        'fields' =>
          {
            'description' => 'This story was posted by a bot.'
          }      
      }
    )
rescue JIRA::HTTPError => e
  puts e.response.code
  puts e.response.message
  puts e.response.body  
end 
```

Result:

```
$ bundle exec ruby jira.rb 
400
Bad Request
<html>
<head><title>400 Bad Request</title></head>
<body bgcolor="white">
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

That mirrors the structure used to create that same issue, with the fields nested under 'fields'.  But obviously it doesn't work.  And it provides absolutely no hint about why it didn't work.

Another attempt:

```
client = JIRA::Client.new(options)
issue = client.Issue.find('ENG-23')
begin
  issue.save!(
      {
        'description' => 'This story was posted by a bot.'
      }
    )
rescue JIRA::HTTPError => e
  puts e.response.code
  puts e.response.message
  puts e.response.body  
end 
```

Result:

```
$ bundle exec ruby jira.rb 
400
Bad Request
<html>
<head><title>400 Bad Request</title></head>
<body bgcolor="white">
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

Another attempt:

```
client = JIRA::Client.new(options)
issue = client.Issue.find('ENG-23')
issue.set_attrs(
    {
      'fields'=>
        {
          'summary' => 'Hello, World!',
          'description' => 'This story was posted by a bot.',
          'project' => {'id' => 10207}, # ENG -- Engineering,
          'issuetype' => {'id' => 10206} # Story
        }
    }
  )
begin
  issue.save!({})
rescue JIRA::HTTPError => e
  puts e.response.code
  puts e.response.message
  puts e.response.body  
end 
```

Result:

```
$ bundle exec ruby jira.rb 
400
Bad Request
<html>
<head><title>400 Bad Request</title></head>
<body bgcolor="white">
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

Another attempt:

```
client = JIRA::Client.new(options)
issue = client.Issue.find('ENG-23')
begin
  issue.save!(
      {
        'fields'=>
          {
            'summary' => 'Hello, World!',
            'description' => 'This story was posted by a bot.',
            'project' => {'id' => 10207}, # ENG -- Engineering,
            'issuetype' => {'id' => 10206} # Story
          }
      }
    )
rescue JIRA::HTTPError => e
  puts e.response.code
  puts e.response.message
  puts e.response.body  
end 
```

Same result.

The real issue here is that I'm an idiot for wasting so much of my time on trying to make this work when it's obviously never going to work.  Is that correct?  Or can this gem actually update issues in JIRA?

If this gem can't update issues in JIRA then why is it wasting space on GitHub?

### PR DESCRIPTION
**Pull request by legkvla**
#100 issue fix

I have written this pull request as example how to fix this issue.
You can reject it and provide more quality code.

### CODE REVIEW

**Comment by legkvla** [comment]
Only this commit https://github.com/legkvla/jira-ruby/commit/3508d06b92247f55f3b11b354b41eab3f725bc10 really needed.

File: lib/jira/resource/project.rb [hunk 1]
```diff
@@ -26,6 +26,16 @@ def issues(options={})
           client.Issue.build(issue)
         end
       end
+
```
Comment (legkvla) [inline lib/jira/resource/project.rb]: Another great feature I missed:
list of assignable users for project.

**Comment by r3mk0** [comment]
Please merge this!

**Comment by omehegan** [comment]
Please PLEASE merge this!

**Comment by omehegan** [comment]
Oh, this PR doesn't actually fix the problem... ??

**Pull request merged**